### PR TITLE
Vaporizer Redux

### PR DESCRIPTION
**Pull request by Kurfursten**
For months I've been trying to get Traders to sell me vapor salts and it's like pulling teeth. Looks like it's a pretty hard sell to convince someone not interested in botany to do some botany, even when offered thousands of credits.

This is an overhaul to the vaporizer. It still fills the same niche but it's a lot less clunky and since it comes preloaded with salts, it's an easier sell to get Vox to trade it. Additionally, they won't end up accidentally deploying it and end up unable to collapse it, because the interface is less restrictive.

Cunning Vox can still sell a "support plan" or something where they supply the humans with refills if they're into growing plants.

🆑 
* tweak: Industrial vaporizer removed from EngiVend to TraderVend for 100 credits and comes preloaded with 1000 vapor salts.
* tweak: The entire UI is less clunky and access is only required to lock/unlock or dump. Now collapse/uncollapse with altclick()
* iconadd: New icon states for locked versus unlocked and active versus inactive.
* tweak: The vaporizer is no longer wrenchable. Instead, it's simply always anchored when deployed.

### CODE REVIEW

**Comment by Durgen** [comment]
The only problem is now the vaporizer suffers from the RnG nature of the tradervend.

**Comment by ghost** [comment]
Instead of premium, how about it's tried as just part of their general stock they can buy and resell? It has very little application in real rounds so everyone could go see how it pans out as a common first.

**Comment by Kurfursten** [comment]
Yeah sure if people want that

Most of the tradervend premiums are really hot garbage though so I didn't think competition would be an issue.

File: code/game/machinery/atmoalter/vaporizer.dm [hunk 1]
```diff
@@ -109,10 +111,12 @@
 	return ui_interact(user)
 
 /obj/machinery/vaporizer/attackby(obj/item/weapon/W, mob/living/user)
-	..()
-	if(waiting_for_ID && istype(W, /obj/item/weapon/card/id)||istype(W, /obj/item/device/pda) && (emagged || allowed(user)) && !unlocked)
-		unlocked = 1
-		to_chat(user,"<span class='notice'>The interface accepts your verification.</span>")
+	if(istype(W, /obj/item/weapon/card/id)||istype(W, /obj/item/device/pda) && (emagged || allowed(user)))
```
Comment (DamianX) [inline code/game/machinery/atmoalter/vaporizer.dm]: isID and isPDA

File: code/game/machinery/atmoalter/vaporizer.dm [hunk 2]
```diff
@@ -210,12 +222,14 @@
...
-/obj/item/vaporizer/CtrlClick(mob/user)
+/obj/item/vaporizer/AltClick(mob/user)
 	unfold()
 
 /obj/item/vaporizer/proc/unfold()
 	if(!folded)
 		folded = new /obj/machinery/vaporizer(src)
 	folded.forceMove(get_turf(src))
+	folded.power_change()
+	folded.update_icon()
```
Comment (DamianX) [inline code/game/machinery/atmoalter/vaporizer.dm]: the call the `power_change` above will end up calling `update_icon` anyway

**Review by DamianX — COMMENTED** [review]
not a full review, just what I noticed while skimming through

**Comment by Kurfursten** [comment]
What's changed?
* Made the changes @DamianX called for
* Took @Durgen and @SarahJohnson489 suggestions - now a 100 credit item for traders

File: code/game/machinery/atmoalter/vaporizer.dm [hunk 3]
```diff
@@ -210,12 +222,13 @@
 /obj/item/vaporizer/attack_self(mob/user)
 	unfold()
 
-/obj/item/vaporizer/CtrlClick(mob/user)
+/obj/item/vaporizer/AltClick(mob/user)
 	unfold()
```
Comment (Kurfursten) [inline code/game/machinery/atmoalter/vaporizer.dm]: add checks for adjacency/alive/etc

**Comment by Kurfursten** [comment]
Now checks for adjacency/alive/etc on altclick

**Pull request merged**